packages/zwave-js/src/lib/commandclass/DoorLockLoggingCC.ts

```diff
@@ -0,0 +1,662 @@
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+		validatePayload(this.payload.length >= 1);
+
+		this.supportedRecordsNumber = this.payload[0];
+		this.persistValues();
+	}
+
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyUInt8,
+		label: "Max logging records stored",
+	})
+	public readonly supportedRecordsNumber: number;
+
+	public toLogEntry(): MessageOrCCLogEntry {
+		return {
+			...super.toLogEntry(),
+			message: {
+				"supported records number": this.supportedRecordsNumber,
+			},
+		};
+	}
+
+	public persistValues(): boolean {
+		if (!super.persistValues()) return false;
+
+		this.getValueDB().setValue(
+			getDoorLockLoggingRecordsSupportedValueID(this.endpointIndex),
+			this.supportedRecordsNumber,
+		);
+
+		return true;
+	}
+}
+
+const convertEventTypeToLabel = (eventType: EventType): string => {
+	return eventTypeLabel[EventType[eventType] as keyof typeof EventType];
+};
+
+@CCCommand(DoorLockLoggingCommand.RecordsSupportedGet)
+@expectedCCResponse(DoorLockLoggingCCRecordsSupportedReport)
+export class DoorLockLoggingCCRecordsSupportedGet extends DoorLockLoggingCC {}
+
+@CCCommand(DoorLockLoggingCommand.RecordReport)
+export class DoorLockLoggingCCRecordReport extends DoorLockLoggingCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+		validatePayload(this.payload.length >= 11);
+
+		this.recordNumber = this.payload[0];
+		this.recordStatus = this.payload[5] >>> 5;
+		if (this.recordStatus === RecordStatus.Empty) {
+			this.record = undefined;
+			this.recordTimestamp = undefined;
+			this.recordEventType = undefined;
+			this.recordUserID = undefined;
+			this.recordUserCode = undefined;
+		} else {
+			const dateSegments = {
+				year: this.payload.readUInt16BE(1),
+				month: this.payload[3],
+				day: this.payload[4],
+				hour: this.payload[5] & 0b11111,
+				minute: this.payload[6],
+				second: this.payload[7],
+			};
+
+			const eventType = this.payload[8];
+			this.recordUserID = this.payload[9];
+
+			// TODO: Parse User Code. The door lock I used does not provide
+			// 	     this data.
+			const userCode = undefined;
+
+			this.record = {
+				eventType: eventType,
+				label: convertEventTypeToLabel(eventType),
+				recordNumber: this.recordNumber,
+				timestamp: segmentsToDate(dateSegments).toISOString(),
+				userId: this.payload[9],
+				userCode,
+			};
+			this.recordTimestamp = segmentsToDate(dateSegments).toISOString();
+			this.recordEventType = eventType;
+			this.recordUserCode = undefined;
+		}
+
+		this.persistValues();
+	}
+
+	public readonly record: Record | undefined;
+	public readonly recordNumber: number;
+	public readonly recordStatus: RecordStatus;
+	public readonly recordTimestamp: string | undefined;
+	public readonly recordEventType: EventType | undefined;
+	public readonly recordUserID: number | undefined;
+	public readonly recordUserCode: string | Buffer | undefined;
```

Comment:
I don't really like duplicating the data here, but let's solve the other issues first.